packages/ldap-integration/service.ts

```diff
@@ -0,0 +1,144 @@
+
+const httpsOptions = {
+  key: fs.readFileSync((process.env.SSL_KEY) ? `${process.env.SSL_KEY}` : ''),
+  cert: fs.readFileSync((process.env.SSL_CERT) ? `${process.env.SSL_CERT}` : '')
+};
+
+const { port = 8081 } = { port : process.env.PORT };
+
+/**
+ * The Ldap
+ *
+ * @class Ldap
+ */
+export class Ldap {
+
+  /**
+   * The express application.
+   * @type {Application}
+   */
+  public app: any;
+  public server: any;
+  /**
+   * Bootstrap the application.
+   *
+   * @class Ldap
+   * @method bootstrap
+   * @static
+   */
+  public static bootstrap(): Ldap {
+    return new Ldap();
+  }
+
+  /**
+   * Constructor.
+   *
+   * @class Ldap
+   * @constructor
+   */
+  constructor() {
+    // configure expressjs application
+    this.middleware();
+
+    // add routes
+    this.routes();
+  }
+
+  /**
+   * Configure application
+   *
+   * @class Ldap
+   * @method middleware
+   */
+  public middleware() {
+    this.app = express();
+
+    // morgan middleware to log HTTP requests
+    this.app.use(logger('":method :url :status :res[content-length] - :response-time ms :referrer :user-agent"'));
+
+    // use json form parser middleware
+    this.app.use(bodyParser.json({ limit: 1024 * 1024 * 20, type: 'application/json' }));
+
+    // use query string parser middleware
+    this.app.use(bodyParser.urlencoded({
+      extended: true
+    }));
+
+    //  catch 404 and forward to error handler
+    this.app.use(function (err: any, req: express.Request, res: express.Response, next: express.NextFunction) {
+      err.status = 404;
+      next(err);
+    });
+
+    // Mount cookie parser
+    this.app.use(cookieParser(process.env.COOKIE_PARSER));
+
+    // Mount override
+    this.app.use(methodOverride());
+
+    // Error handling
+    this.app.use(errorHandler());
+
+    this.app.set('port', port);
+
+    this.server = https.createServer(httpsOptions, this.app);
+
+    /**
+     * Listen on provided port, on all network interfaces.
+     */
+    this.server.listen(port, () => {
+      console.log(`Server listening on ${port}`);
+    });
+  }
+
+  /**
+   * Create REST API routes
+   *
+   * @class Ldap
+   * @method routes
+   */
+  public routes() {
+
+    const router = express.Router();
+
+    // configure CORS
+    const corsOptions: cors.CorsOptions = {
+      allowedHeaders: ['Origin', 'X-Requested-With', 'Content-Type', 'Accept', 'X-Access-Token'],
+      credentials: true,
+      methods: 'GET,HEAD,OPTIONS,PUT,PATCH,POST,DELETE',
+      origin: '*',
+      preflightContinue: false
+    };
+
+    router.use(cors(corsOptions));
+
+    // root request
+    router.get('/', (req: express.Request, res: express.Response, next: express.NextFunction) => {
+      res.json({ message: `One Platform Ldap microservice APIs.` });
+      next();
+    });
+
+    // create API routes
+    this.app.use('/', router);
+    this.app.use('/ldap', LdapAPIRouter);
+
+    // enable CORS pre-flight
+    router.options('*', cors(corsOptions));
+  }
+}
+
```

Comment:
@1-Platform/one-platform-developers I have a feeling that it might be better if we can break the middleware and the router can be broken into separate files. What do you folks think?